src/index.ts

```diff
@@ -136,11 +90,19 @@ export async function init(config: IClientConfig): Promise<EppoClient> {
 /**
  * Used to access a singleton SDK client instance.
  * Use the method after calling init() to initialize the client.
- * @returns a singleton client instance
+ * @returns a singleton client instance or throws an Error if init() has not been called
  */
 export function getInstance(): EppoClient {
   if (!clientInstance) {
     throw Error('Expected init() to be called to initialize a client instance');
   }
   return clientInstance;
 }
+
+function newEventDispatcher(sdkKey: string) {
+  const eventQueue = new FileBackedNamedEventQueue('events');
+  const emptyNetworkStatusListener =
+    // eslint-disable-next-line @typescript-eslint/no-empty-function
+    { isOffline: () => false, onNetworkStatusChange: () => {} };
```

Comment:
i opted to not provide an implementation for `NetworkStatusListener` since I didn't think it made sense for node server side. We could expose an optional property in `IClientConfig` though, to allow users to provide their own implementation. thoughts?

Reply:
if the server is offline, then nothing else will work, so it feels like pointless to handle that, except in the client side